src/main/java/appeng/core/sync/packets/PacketCompressedNBT.java

```diff
@@ -28,93 +28,108 @@
 import appeng.core.sync.AppEngPacket;
-import appeng.core.sync.network.INetworkInfo;
+import appeng.core.sync.AppEngPacketHandler;
+import appeng.helpers.Reflected;
 
 
-public class PacketCompressedNBT extends AppEngPacket
+public class PacketCompressedNBT implements AppEngPacket, AppEngPacketHandler<PacketCompressedNBT, AppEngPacket>
 {
 
 	// input.
-	final NBTTagCompound in;
+	private NBTTagCompound in;
 	// output...
-	private final ByteBuf data;
-	private final GZIPOutputStream compressFrame;
-	int writtenBytes = 0;
-	boolean empty = true;
+	private ByteBuf data;
+	private GZIPOutputStream compressFrame;
 
-	// automatic.
-	public PacketCompressedNBT( final ByteBuf stream ) throws IOException
+	@Reflected
+	public PacketCompressedNBT()
+	{
+		// automatic.
+	}
+
+	// api
+	public PacketCompressedNBT( final NBTTagCompound din )
+	{
+		this.in = din;
+	}
+
+	@Override
+	public void fromBytes( final ByteBuf buf )
 	{
 		this.data = null;
 		this.compressFrame = null;
 
-		final GZIPInputStream gzReader = new GZIPInputStream( new InputStream()
+		final GZIPInputStream gzReader;
+		try
 		{
-
-			@Override
-			public int read() throws IOException
+			gzReader = new GZIPInputStream( new InputStream()
 			{
-				if( stream.readableBytes() <= 0 )
+
+				@Override
+				public int read() throws IOException
 				{
-					return -1;
+					if( buf.readableBytes() <= 0 )
+					{
+						return -1;
+					}
+
+					return buf.readByte() & 0xff;
 				}
+			} );
 
-				return stream.readByte() & 0xff;
-			}
-		} );
+			final DataInputStream inStream = new DataInputStream( gzReader );
+			this.in = CompressedStreamTools.read( inStream );
+			inStream.close();
+		}
+		catch( final IOException e )
+		{
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
 
-		final DataInputStream inStream = new DataInputStream( gzReader );
-		this.in = CompressedStreamTools.read( inStream );
-		inStream.close();
 	}
 
-	// api
-	public PacketCompressedNBT( final NBTTagCompound din ) throws IOException
+	@Override
+	public void toBytes( final ByteBuf buf )
 	{
-
-		this.data = Unpooled.buffer( 2048 );
-		this.data.writeInt( this.getPacketID() );
-
-		this.in = din;
-
-		this.compressFrame = new GZIPOutputStream( new OutputStream()
+		try
 		{
-
-			@Override
-			public void write( final int value ) throws IOException
+			this.compressFrame = new GZIPOutputStream( new OutputStream()
 			{
-				PacketCompressedNBT.this.data.writeByte( value );
-			}
-		} );
-
-		CompressedStreamTools.write( din, new DataOutputStream( this.compressFrame ) );
-		this.compressFrame.close();
 
-		this.configureWrite( this.data );
+				@Override
+				public void write( final int value ) throws IOException
+				{
+					PacketCompressedNBT.this.data.writeByte( value );
+				}
+			} );
+			CompressedStreamTools.write( this.in, new DataOutputStream( this.compressFrame ) );
+			this.compressFrame.close();
+		}
+		catch( final IOException e )
+		{
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00108)